llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -421,27 +466,132 @@ Expected<SectionRef> SectionRef::create(MCCASBuilder &MB,
+      if (!Size)
+        return Size.takeError();
+      if (*Size + RelocDataIndex > RelocationData.size()) {
+        return createStringError(inconvertibleErrorCode(),
+                                 "Relocation Data not encoded correctly, "
+                                 "reading past the end of the relocation data");
+      }
+      for (uint32_t I = 0; I < *Size; I++)
+        SectionData[Relocs.r_word0 + I] = RelocationData[RelocDataIndex + I];
+      RelocDataIndex += *Size;
+    }
+    return SectionData;
+  }
+  return createStringError(
+      inconvertibleErrorCode(),
+      "First reference to debug line section must be a RelocAddendRef object!");
+}
+
+Expected<uint64_t> MCCASReader::materializeDebugLineSection(
+    const SmallVector<cas::ObjectRef> &Refs, StringRef RelocData) {
+  bool IsPaddingRefPresent = false;
+  auto DebugLineData =
+      applyRelocationAddends(Refs, RelocData, DebugLineRef::KindString,
+                             "debug_line", IsPaddingRefPresent);
+  if (!DebugLineData)
+    return DebugLineData.takeError();
+  OS << *DebugLineData;
+  if (IsPaddingRefPresent) {
+    auto PadRef =
+        MCObjectProxy::get(Schema, Schema.CAS.getProxy(Refs[Refs.size() - 1]));
+    if (!PadRef)
+      return PadRef.takeError();
+    if (auto P = PaddingRef::Cast(*PadRef)) {
+      auto PadSize = P->materialize(OS);
+      if (!PadSize)
+        return PadSize.takeError();
+      return *PadSize + DebugLineData->size();
+    }
+  }
+  return DebugLineData->size();
+}
+
+Expected<bool>
+MCCASReader::isDebugLineSection(const SmallVector<cas::ObjectRef> &Refs) {
+  // The debug_line section must have at least 2 references, one for the
+  // relocation addends and the other for the debug_line section contents.
+  if (Refs.size() < 2)
+    return false;
+  auto RelocRef = MCObjectProxy::get(Schema, Schema.CAS.getProxy(Refs[0]));
+  if (!RelocRef)
+    return RelocRef.takeError();
+  if (auto R = RelocAddendRef::Cast(*RelocRef)) {
+    auto DbgLineRef = MCObjectProxy::get(Schema, Schema.CAS.getProxy(Refs[1]));
+    if (!DbgLineRef)
+      return DbgLineRef.takeError();
+    if (auto D = DebugLineRef::Cast(*DbgLineRef))
+      return true;
+  }
+  return false;
+}
+
 Expected<uint64_t> SectionRef::materialize(MCCASReader &Reader) const {
   // Start a new section for relocations.
   Reader.Relocations.emplace_back();
 
-  unsigned Size = 0;
   StringRef Remaining = getData();
   auto Refs = decodeReferences(*this, Remaining);
   if (!Refs)
     return Refs.takeError();
 
-  for (auto ID : *Refs) {
-    auto FragmentSize = Reader.materializeSection(ID);
-    if (!FragmentSize)
-      return FragmentSize.takeError();
-    Size += *FragmentSize;
-  }
+  auto IsDebugLineSection = Reader.isDebugLineSection(*Refs);
+  if (!IsDebugLineSection)
+    return IsDebugLineSection.takeError();
+
+  if (*IsDebugLineSection) {
+    // Handle Debug Line Section
+    return Reader.materializeDebugLineSection(*Refs, Remaining);
+  } else {
```

Comment:
Let's remove the `else`

https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return

Note how the diff will be much smaller, making reviewing easier ;)